Cython/Compiler/ExprNodes.py

```diff
@@ -12212,6 +12212,27 @@ def c_types_okay(self, type1, type2):
         return (type1.is_int or type1.is_enum) \
             and (type2.is_int or type2.is_enum)
 
+    def analyse_pytyping_modifiers(self, env):
```

Comment:
Since these methods are really specific to the `|` operator, I suggest creating a new subtype `BitwiseOrNode` for them. We already do this for several other operators that are more special than others.